Tests/SonarScanner.MSBuild.Shim.Test/PathHelperTests.cs

```diff
@@ -29,164 +30,162 @@ namespace SonarScanner.MSBuild.Shim.Test
+            PathHelper.BestCommonPrefix(null).Should().BeNull();
 
-            // Assert
-            result.Should().BeNull();
-        }
+        [TestMethod]
+        public void BestCommonPrefix_WhenEmpty_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(Enumerable.Empty<DirectoryInfo>()).Should().BeNull();
 
         [TestMethod]
-        public void GetCommonRoot_WhenNoCommonPath_ReturnsNull()
-        {
-            PathHelper.GetCommonRoot(new[]
+        public void BestCommonPrefix_WhenNoCommonPath_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(new[]
             {
-                new DirectoryInfo("C:\\"),
-                new DirectoryInfo("D:\\Foo"),
+                new DirectoryInfo(@"C:\"),
+                new DirectoryInfo(@"D:\Dir"),
             }).Should().BeNull();
-        }
 
         [TestMethod]
-        public void GetCommonRoot_WhenCommonPath_ReturnsTheLongestCommonPart()
-        {
-            PathHelper.GetCommonRoot(new[]
+        public void BestCommonPrefix_WhenNoCommonPath_SameCountInEachGroup_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(new[]
             {
-                new DirectoryInfo("C:\\Foo"),
-                new DirectoryInfo("C:\\Foo\\Bar"),
-                new DirectoryInfo("C:\\Foo\\FooBar"),
-            }).FullName.Should().Be("C:\\Foo");
-        }
+                new DirectoryInfo(@"C:\"),
+                new DirectoryInfo(@"C:\Dir"),
+                new DirectoryInfo(@"D:\DirA"),
+                new DirectoryInfo(@"D:\DirB\SubDir"),
+                new DirectoryInfo(@"Z:\"),
+                new DirectoryInfo(@"Z:\Dir"),
+            }).Should().BeNull();
 
         [TestMethod]
-        public void GetCommonRoot_WhenCommonPathOfFiles_ReturnsTheLongestCommonPart()
-        {
-            PathHelper.GetCommonRoot(new[]
+        public void BestCommonPrefix_WhenNoCommonPath_SameCountInMostCommonGroup_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(new[]
             {
-                new DirectoryInfo("C:\\Foo.cs"),
-                new DirectoryInfo("C:\\Foo\\Bar.cs"),
-                new DirectoryInfo("C:\\Foo\\FooBar.cs"),
-            }).FullName.Should().Be("C:\\");
-        }
+                new DirectoryInfo(@"C:\Temp"),
+                new DirectoryInfo(@"D:\ThreeTimes\A"),
+                new DirectoryInfo(@"D:\ThreeTimes\B"),
+                new DirectoryInfo(@"D:\ThreeTimes\C"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\A"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\B"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\C"),
+            }).Should().BeNull();
 
         [TestMethod]
-        public void GetParts_WhenNull_ThrowsArgumentNullException()
-        {
-            // Arrange
-            // Act
-            Action act = () => PathHelper.GetParts(null);
+        public void BestCommonPrefix_WhenNoCommonPath_ReturnsMostCommonOne_Simple() =>
```

Comment:
```suggestion
        public void BestCommonPrefix_WhenNoCommonPath_Simple_ReturnsMostCommonOne() =>
```

Reply:
Suffixes don't cost on readability and make it easier to see where the difference really is (table-style)
![image](https://user-images.githubusercontent.com/57188685/184126447-cec20abf-25fb-46f4-83e1-56374a15bd79.png)